type-definitions/most.d.ts

```diff
@@ -1,385 +1,788 @@
+
+	export function timestamp<A>(s: Stream<A>): Stream<TimeValue<A>>;
+	export function delay<A>(dt: number, s: Stream<A>): Stream<A>;
+
+	export function fromPromise<A>(p: Promise<A>): Stream<A>;
+	export function await<A>(s: Stream<Promise<A>>): Stream<A>;
+
+	export function sample<A, B, R>(
+		fn: (a: A, b: B) => R,
+		sampler: Stream<any>,
+		a: Stream<A>,
+		b: Stream<B>
+	): Stream<R>;
+	export function sample<A, B, C, R>(
+		fn: (a: A, b: B, c: C) => R,
+		sampler: Stream<any>,
+		a: Stream<A>,
+		b: Stream<B>,
+		c: Stream<C>
+	): Stream<R>;
+	export function sample<A, B, C, D, R>(
+		fn: (a: A, b: B, c: C, d: D) => R,
+		sampler: Stream<any>,
+		a: Stream<A>,
+		b: Stream<B>,
+		c: Stream<C>,
+		d: Stream<D>
+	): Stream<R>;
+	export function sample<A, B, C, D, E, R>(
+		fn: (a: A, b: B, c: C, d: D, e: E) => R,
+		sampler: Stream<any>,
+		a: Stream<A>,
+		b: Stream<B>,
+		c: Stream<C>,
+		d: Stream<D>,
+		e: Stream<E>
+	): Stream<R>;
+
+	export function sampleWith<A>(sampler: Stream<any>, s: Stream<A>): Stream<A>;
+
+	export function zip<A, B, R>(
+		fn: (a: A, b: B) => R,
+		a: Stream<A>,
+		b: Stream<B>
+	): Stream<R>;
+	export function zip<A, B, C, R>(
+		fn: (a: A, b: B, c: C) => R,
+		a: Stream<A>,
+		b: Stream<B>,
+		c: Stream<C>
+	): Stream<R>;
+	export function zip<A, B, C, D, R>(
+		fn: (a: A, b: B, c: C, d: D) => R,
+		a: Stream<A>,
+		b: Stream<B>,
+		c: Stream<C>,
+		d: Stream<D>
+	): Stream<R>;
+	export function zip<A, B, C, D, E, R>(
+		fn: (a: A, b: B, c: C, d: D, e: E) => R,
+		a: Stream<A>,
+		b: Stream<B>,
+		c: Stream<C>,
+		d: Stream<D>,
+		e: Stream<E>
+	): Stream<R>;
+	export function recoverWith<A, B>(p: (a: B) => A, s: Stream<A>): Stream<A>;
+	export function throwError(e: Error): Stream<any>;
+
+	export function multicast<A>(s: Stream<A>): Stream<A>;
+}
+
+declare module "most/src/scheduler/ClockTimer" {
+	import { Task, Timer } from 'most';
+	class ClockTimer implements Timer {
+		now(): number;
+		setTimer(f: () => any, dt: number): Task | number;
+		clearTimer(task: Task | number): void;
+	}
+
+	export default ClockTimer;
+}
+
+declare module "most/lib/scheduler/ClockTimer" {
+	import ClockTimer from 'most/src/scheduler/ClockTimer';
+
+	export = ClockTimer;
+}
+
+declare module "most/src/scheduler/defaultScheduler" {
+	import { Scheduler } from 'most';
+	const defaultScheduler: Scheduler;
+	export default defaultScheduler;
+}
+
+declare module "most/lib/scheduler/defaultScheduler" {
+	import defaultScheduler from 'most/src/scheduler/defaultScheduler';
+	export = defaultScheduler;
+}
+
+declare module "most/src/scheduler/PropagateTask" {
+	import { Task, Sink } from 'most';
+	class PropagateTask<T> implements Task {
+		public active: boolean;
+		constructor(run: (t: number, x: T, sink: Sink<T>) => any, value: T, sink: Sink<T>);
+		static event<T>(x: T, sink: Sink<T>): PropagateTask<T>;
+		static end<T>(x: T, sink: Sink<T>): PropagateTask<T>;
+		static error(err: Error, sink: Sink<any>): PropagateTask<any>;
```

Comment:
I think TypeScript is poor in handling this, but when you go to write anything with this you can never expect any type of value but an error to be propagated, however to satisfy interfaces you will have to write `.event` and `.end` on a `Sink` and I felt it was poor to force any types there.